### PR TITLE
feat: add tool list subcommand

### LINKED ISSUE #24: [Umbrella] Add `tool` subcommand
**Please provide an in-depth description of the question you have**:

According to the `issues` discussion, add the `tool` subcommand to output information such as the version of tools, and download tools.

task:

- [ ]  add `tool list` subcommand , output toos information.
- [ ]  add `tool install` subcommand, download tools.

### PR DESCRIPTION
**Pull request by prodanlabs**
Signed-off-by: prodan <pengshihaoren@gmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
List the version and other information of the cli tool of the component

**Which issue(s) this PR fixes**:
Fixes #24 

**Special notes for your reviewer**:
```
/opt/workspaces/kurator$ ./out/linux-amd64/kurator tool list -h
list tool information

Usage:
  kurator tool list [flags]

Aliases:
  list, ls

Examples:
  # List all components Cli tool.
  kurator tool list

  # List the Cli tools for all components and more information (such as tool path).
  kurator tool list -o wide

  # List specified components Cli tool.
  kurator tool list istio karmada

  # List component Cli tools in JSON output format.
  kurator tool list -o json

  # List component Cli tools in YAML output format.
  kurator tool list -o yaml

  # List a single components Cli tool in JSON output format.
  kurator tool list istio -o json


Flags:
  -h, --help            help for list
  -o, --output string   Output format. (-o|--output=)[table wide json yaml] (default "table")

Global Flags:
      --context string           name of the kubeconfig context to use
      --dry-run                  console/log output only, make no changes.
      --home-dir string          install path, default to $HOME/.kurator (default "/home/prodan/.kurator")
  -c, --kubeconfig string        path to the kubeconfig file, default to karmada apiserver config (default "/etc/karmada/karmada-apiserver.config")
      --wait-interval duration   interval used for checking pod ready, default value is 1s. (default 1s)
      --wait-timeout duration    timeout used for checking pod ready, default value is 2m. (default 2m0s)
```
table output
```
/opt/workspaces/kurator$ ./out/linux-amd64/kurator tool list 
NAME            VERSION STATUS       
istio           1.13.3  Ready        
karmada         v1.2.1  NotDownloaded
kubeedge        1.10.0  NotDownloaded
submariner      v0.12.2 NotDownloaded
/opt/workspaces/kurator$ ./out/linux-amd64/kurator tool list -o wide
NAME            VERSION STATUS          CLI                                             HUB                     RELEASE-URL-PREFIX                                         
istio           1.13.3  Ready           /home/prodan/.kurator/istio/1.13.3/istioctl     docker.io/istio         https://github.com/istio/istio/releases/download           
karmada         v1.2.1  NotDownloaded                                                                           https://github.com/karmada-io/karmada/releases/download    
kubeedge        1.10.0  NotDownloaded                                                   docker.io/kebeedge      https://github.com/kubeedge/kubeedge/releases/download     
submariner      v0.12.2 NotDownloaded                                                                           https://github.com/submariner-io/releases/releases/download
```
json output
```
/opt/workspaces/kurator$ ./out/linux-amd64/kurator tool list istio karmada -o json
{
        "istio": {
                "Cli": "/home/prodan/.kurator/istio/1.13.3/istioctl",
                "Hub": "docker.io/istio",
                "Name": "istio",
                "ReleaseURLPrefix": "https://github.com/istio/istio/releases/download",
                "Status": "Ready",
                "Version": "1.13.3"
        },
        "karmada": {
                "Cli": "",
                "Hub": "",
                "Name": "karmada",
                "ReleaseURLPrefix": "https://github.com/karmada-io/karmada/releases/download",
                "Status": "NotDownloaded",
                "Version": "v1.2.1"
        }
}
```
yaml output
```
/opt/workspaces/kurator$ ./out/linux-amd64/kurator tool list istio karmada -o yaml
istio:
  Cli: /home/prodan/.kurator/istio/1.13.3/istioctl
  Hub: docker.io/istio
  Name: istio
  ReleaseURLPrefix: https://github.com/istio/istio/releases/download
  Status: Ready
  Version: 1.13.3
karmada:
  Cli: ""
  Hub: ""
  Name: karmada
  ReleaseURLPrefix: https://github.com/karmada-io/karmada/releases/download
  Status: NotDownloaded
  Version: v1.2.1
```

**Does this PR introduce a user-facing change?**:

```release-note
add `kurator tool list` subcommand
```

### CODE REVIEW

File: pkg/tool/list.go [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+	case "kubeedge":
+		return "keadm"
+	case "argocd":
+		return "argocd"
+	default:
+		return ""
+	}
+}
+
+func isReady(path string) (string, string) {
```
Comment (zirain) [inline pkg/tool/list.go:298]: prefer to return a boolean

File: cmd/kurator/app/tool/tool.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/spf13/cobra"
+
+	"kurator.dev/kurator/pkg/generic"
+	"kurator.dev/kurator/pkg/tool"
+)
+
+func NewCmd(opts *generic.Options) *cobra.Command {
+	toolCmd := &cobra.Command{
+		Use:                   "tool",
+		Short:                 "Cil tool information for the component",
```
Comment (zirain) [inline cmd/kurator/app/tool/tool.go:31]: nit: Cli

File: pkg/tool/list.go (hunk 1 above)
Comment (prodanlabs) [reply pkg/tool/list.go:298]: thanks for your review.
The name is confusing, i change the method name .

File: go.mod [hunk 3]
```diff
@@ -5,6 +5,7 @@ go 1.17
 require (
 	github.com/Masterminds/sprig/v3 v3.2.2
 	github.com/bramvdbogaerde/go-scp v1.2.0
+	github.com/gosuri/uitable v0.0.4
```
Comment (hzxuzhonghu) [inline go.mod:8]: My doubt is that is this package still maintained.

**Review by hzxuzhonghu — APPROVED** [review]
LGTM

File: pkg/tool/list.go [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+	if !exists(path) {
+		return "NotDownloaded", ""
```
[the remark this answers is not in the document]
Comment (prodanlabs) [reply pkg/tool/list.go]: done

File: cmd/kurator/app/tool/tool.go (hunk 2 above)
Comment (prodanlabs) [reply cmd/kurator/app/tool/tool.go]: done

**Review by zirain — APPROVED** [review]
/lgtm

**Pull request merged**